server/src/main/java/com/thoughtworks/go/server/dao/PluginSqlMapDao.java

```diff
@@ -113,4 +113,23 @@ protected void doInTransactionWithoutResult(TransactionStatus status) {
             }
         });
     }
+
+    @Override
+    public void deletePluginIfExists(String pluginId) {
+        String cacheKey = cacheKeyForPluginSettings(pluginId);
+        Plugin plugin = this.findPlugin(pluginId);
+
+        if (plugin instanceof NullPlugin) {
+            return;
+        }
+
+        synchronized (cacheKey) {
+            transactionTemplate.execute((TransactionCallback) transactionStatus -> sessionFactory.getCurrentSession()
+                    .createQuery(String.format("delete from %s where pluginId=:pluginId", Plugin.class.getSimpleName()))
```

Comment:
Couldn't we just do `sessionFactory.getCurrentSession().delete(plugin);`

Reply:
`transactionTemplate.execute` method expects a return value and `sessionFactory.getCurrentSession().delete` returns a void. This requires adding a new execute method which supports void return type.